### DISCUSSION TITLE
Memory error, anything I can do or just need better G-card?

### DISCUSSION THREAD

**Discussion by Two82** [Comfy-Org/ComfyUI#1928] [Q&A]
Firstly, thank you for taking the time to read this post. Your help is much appreciated. 

I am assuming this is just a case of my 3060 not being able to handle the load. However, I am not real versed in py script, until now I have always been able to solve all my own issues in blender, SD, etc., by referencing py manual, stack overflow, or a GitHub preexisting issue. I just can't seem to find much about comfy optimization. Unlike Kohya and auto1111 where there seemed to be plenty.

 My machine with 12th Gen I7-12700 with RTX 3060 and 32 gb ram has been able to handle pretty much anything i have thrown at it so far even if it took a lot of optimization tweaks and long render times. So, I was hoping maybe my inexperience was the issue here, not my lack of a decent GPU. Even though that is an issue either way. So, I thought I would reach out to those who are more knowledgeable than myself.

I don't like asking for help, I find solving myself through research is better for learning. So, if my format is incorrect, I apologize in advance.

my workflow 
![workflow](https://github.com/comfyanonymous/ComfyUI/assets/126403375/1dc480fe-5371-44d4-9e50-9799a3451079)

cmd line error
Prompt executed in 244.37 seconds
got prompt
Using pytorch attention in VAE
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
Using pytorch attention in VAE
Leftover VAE keys ['model_ema.decay', 'model_ema.num_updates']
ERROR:root:!!! Exception during processing !!!
ERROR:root:Traceback (most recent call last):
  File "C:\ComfyUI_windows_portable_nvidia_cu121_or_cpu\ComfyUI_windows_portable\ComfyUI\execution.py", line 153, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ComfyUI_windows_portable_nvidia_cu121_or_cpu\ComfyUI_windows_portable\ComfyUI\execution.py", line 83, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ComfyUI_windows_portable_nvidia_cu121_or_cpu\ComfyUI_windows_portable\ComfyUI\execution.py", line 76, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ComfyUI_windows_portable_nvidia_cu121_or_cpu\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI_IPAdapter_plus\IPAdapterPlus.py", line 509, in preprocess
    clip_embed = clip_vision.encode_image(image)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ComfyUI_windows_portable_nvidia_cu121_or_cpu\ComfyUI_windows_portable\ComfyUI\comfy\clip_vision.py", line 43, in encode_image
    pixel_values = clip_preprocess(image.to(self.load_device))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ComfyUI_windows_portable_nvidia_cu121_or_cpu\ComfyUI_windows_portable\ComfyUI\comfy\clip_vision.py", line 16, in clip_preprocess
    image = torch.nn.functional.interpolate(image.movedim(-1, 1), size=(round(scale * image.shape[1]), round(scale * image.shape[2])), mode="bicubic", antialias=True)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\ComfyUI_windows_portable_nvidia_cu121_or_cpu\ComfyUI_windows_portable\python_embeded\Lib\site-packages\torch\nn\functional.py", line 4027, in interpolate
    return torch._C._nn._upsample_bicubic2d_aa(input, output_size, align_corners, scale_factors)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
torch.cuda.OutOfMemoryError: Allocation on device 0 would exceed allowed memory. (out of memory)
Currently allocated     : 8.90 GiB
Requested               : 4.54 GiB
Device limit            : 12.00 GiB
Free (according to CUDA): 0 bytes
PyTorch limit (set by user-supplied memory fraction)
                        : 17179869184.00 GiB

**Comment by iamwavecut** [comment]
--lowvram

**Comment by JasonS09** [comment]
Did you manage to solve this issue? I'm experiencing the same. I don't think --lowvram is the solution since it claims it's automatically activated at low vram detection and I have never needed to use the flag.

**Reply by Two82** [reply]
I updated comfy via update.bat and switched my loaders and samplers to the ones found in the default folders that are named loaders etc. instead of 3rd party nodes, defiantly helped, I also started using T2i sdxl instead of controlnet sdxl as they are much lighter. Also starting to play with the lcm models, but there is a loss of quality, and they are very touchy to any small attribute change.
